godot-core/src/builtin/file.rs

```diff
@@ -0,0 +1,459 @@
+                    &path,
+                    flags,
+                    FileAccess::get_open_error()
+                ),
+            )
+        })?;
+
+        Ok(Self::init(fa))
+    }
+
+    /// Opens a file encrypted with a `pass` located at `path`.
+    /// For [ModeFlags] description check the [GFile] documentation
+    pub fn open_encrypted_with_pass(
+        path: GodotString,
+        flags: ModeFlags,
+        pass: GodotString,
+    ) -> std::io::Result<Self> {
+        let fa =
+            FileAccess::open_encrypted_with_pass(path.clone(), flags, pass).ok_or_else(|| {
+                IoError::new(
+                    ErrorKind::Other,
+                    format!(
+                        "Can't open file {} in mode {:?}; GodotError: {:#?}",
+                        &path,
+                        flags,
+                        FileAccess::get_open_error()
+                    ),
+                )
+            })?;
+
+        Ok(Self::init(fa))
+    }
+
+    /// Closes the inner [Gd]<[FileAccess]> and destroys the [GFile] instance
+    pub fn close(mut self) {
+        self.fa.close()
+    }
+
+    /// Deconstructs the [GFile] intance and retrieves the inner [Gd]<[FileAccess]>
+    pub fn into_inner(self) -> Gd<FileAccess> {
+        self.fa
+    }
+
+    // Remaps of the internal FileAccess methods
+
+    // Reads
+
+    /// Reads the next 8 bits from the file as [u8]. Exposed Godot method.
+    pub fn get_8(&self) -> IoResult<u8> {
+        let val = self.fa.get_8();
+        self.check_error()?;
+        Ok(val)
+    }
+
+    /// Reads the next 16 bits from the file as [u16].
+    /// For details check the [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html)
+    pub fn get_16(&self) -> IoResult<u16> {
+        let val = self.fa.get_16();
+        self.check_error()?;
+        Ok(val)
+    }
+
+    /// Reads the next 32 bits from the file as [u32]. Exposed Godot method.
+    /// For details check the [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html)
+    pub fn get_32(&self) -> IoResult<u32> {
+        let val = self.fa.get_32();
+        self.check_error()?;
+        Ok(val)
+    }
+
+    /// Reads the next 64 bits from the file as [u64]. Exposed Godot method.
+    /// For details check the [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html)
+    pub fn get_64(&self) -> IoResult<u64> {
+        let val = self.fa.get_64();
+        self.check_error()?;
+        Ok(val)
+    }
+
+    /// Reads the next 64 bits from the file as [u64]. Exposed Godot method.
+    /// For details check the [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html)
+    pub fn get_pascal_string(&mut self) -> IoResult<GodotString> {
+        let val = self.fa.get_pascal_string();
+        self.check_error()?;
+        Ok(val)
+    }
```

Comment:
This method in Godot is very confusing. What is a "pascal string"?
Also, the 64 bytes are not documented anywhere, can we rely on this?

Do you know what's the use case of this?

Reply:
Also nitpick, please end all sentences with punctuation, both in docs and code comments.

Reply:
@Bromeon This remark about 64 bytes is omission, as I've made this PR mid writing and polishing documentation (just copy-paste artefact) :)

This pascal string is a string that is readible and saveable from/into binary files without encoding it. I use it myself for writing the `godot_class` (struct name) and `uid` as a _header_ of serialized rust resources - can be fast and easily retrieved from binary file, without making global assumption how long the binary representation should be. 

You can `write_string` in Godot FileAccess, but can't retrieve _one string_ as its not delimited in any way from other contents. You can with pascal string (though I didn't test it in text files, only binary).

Reply:
> This pascal string is a string that is readible and saveable from/into binary files without encoding it. I use it myself for writing the `godot_class` (struct name) and `uid` as a _header_ of serialized rust resources - can be fast and easily retrieved from binary file, without making global assumption how long the binary representation should be.

The crucial information here is that it stores a length prefix, unlike C strings, that store a null terminator.

There seems to be [a Wikipedia article](https://en.wikipedia.org/wiki/String_(computer_science)#Length-prefixed), maybe we could link to this?